tools/prepare_dataset.py

```diff
@@ -0,0 +1,224 @@
+import json
+import os
+from functools import cached_property
+from multiprocessing import Pool
+from pathlib import Path
+
+import numpy as np
+from datasets import load_dataset, load_from_disk
+from torch import distributed as dist
+from tqdm import tqdm
```

Comment:
Not in the requirements. Not sure we need it, but if really needed you'll need to add it to optional requirements.